interfaces/oauth_client/v0/README.md

```diff
@@ -0,0 +1,86 @@
+# `oauth_client`
+
+## Usage
+
+This relation interface describes the expected behavior of any charm claiming to be able to interface with an OAuth2/OIDC Provider. Charms claiming to provide this interface must implement the OAuth2/OIDC protocol and register a client for each relation.
+
+In most cases, this will be accomplished using the oauth_client library, although charm developers are free to provide alternative libraries as long as they fulfill the behavioral and schematic requirements described in this document.
+
+## Direction
+
+The `oauth_client` interface implements a provider/requirer pattern.
+The consumer is a charm that wishes to act as a oauth2 client, and the provider is a charm exposing an OAuth2/OIDC provider.
+
+```mermaid
+flowchart TD
+    Requirer -- redirect_uri, audience, scope, grant_types, token_endpoint_auth_method --> Provider
+    Provider -- issuer_url, authentication_endpoint, token_endpoint, userinfo_endpoint, introspection_endpoint, jwks_endpoint, client_id, client_secret_id, scope, groups, ca_chain --> Requirer
+```
+
+## Behavior
+
+The requirer and the provider MUST adhere to a certain set of criteria to be considered compatible with the interface.
+
+### Provider
+
+- MUST provide its endpoints(authentication_endpoint, token_endpoint, userinfo_endpoint, introspection_endpoint and jwks_endpoint) in the relation databag.
+- MUST provide the issuer_url in the relation databag.
+- MUST provide the supported scopes in the databag.
+- MUST register a client and provide the client_id in the databag.
+- MUST register a client, place the client_secret in a juju secret and provide the juju secret ID in the databag (client_secret_id).
```

Comment:
This means it wont be compatible with Juju 2.9. Is that an OK tradeoff? I think if so, this should be mentioned somewhere.

Reply:
Since we are starting from scratch with IAM, I don't know why we would want to support juju 2.x.

Reply:
I agree it should be mentioned somewhere to make it super clear